### DISCUSSION TITLE
rg -g exits with error code and no output

### DISCUSSION THREAD

**Discussion by gajus** [BurntSushi/ripgrep#2382] [General] [answered]
```
rg 'test' .
```

^This works

```
rg -g '*.js' .
```

This exits with exit status 1 and no other output.

I tried adding `--debug`:

```
DEBUG|globset|/usr/share/cargo/registry/ripgrep-12.1.1/debian/cargo_registry/globset-0.4.5/src/lib.rs:431: built glob set; 0 literals, 0 basenames, 12 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
DEBUG|globset|/usr/share/cargo/registry/ripgrep-12.1.1/debian/cargo_registry/globset-0.4.[5](https://github.com/contra/gaia/actions/runs/3809007096/jobs/6480049073#step:7:6)/src/lib.rs:431: built glob set; 0 literals, 0 basenames, 1 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
```

The output just ends here with exit 1.

For context, I just realized that this is only happening when rg is evaluated by GitHub worker.

<img width="1076" alt="Screen Shot 2022-12-30 at 3 06 02 PM" src="https://user-images.githubusercontent.com/973543/210111634-ecf44dfc-0a42-471e-8950-ac87f4063cc0.png">

<img width="1686" alt="Screen Shot 2022-12-30 at 3 08 05 PM" src="https://user-images.githubusercontent.com/973543/210111740-111423bf-5f75-4f97-8a9d-3564b402d681.png">

ripgrep v12.1.1

I cannot tell if it is a bug or if I am overlooking something obvious.

**Comment by BurntSushi** [comment, marked as answer]
No output means nothing was found. Do you expect something to be found? Try `-uuu` to disable all smart filtering.

And are you sure you've shown all debug output? Doesn't look like it.

**Reply by gajus** [reply]
For context, I am running this in GitHub action worker.

I just realized that if I ssh into the worker and run the equivalent command, it works as expected.

It appears that this has something to do with how GitHub CI executes shell commands.

**Reply by gajus** [reply]
Updated original question with a screenshot of the "full" output.

**Reply by BurntSushi** [reply]
Yeah that's important context....... Your first command specifies the directory to search. Your second command doesn't. So give the directory to your second command.

The reason why running it interactively works is because ripgrep tries to detect whether stdin is attacged to it. GitHub Actions sets up their environment such that it looks like stdin is active, so ripgrep searches that when you don't provide a directory to search.

Why does GitHub Actions do this? I don't know.

**Reply by gajus** [reply]
What do you mean by:

> So give the directory to your second command.

As far as I can tell, all commands specify the directory (the dot at the end). Do you mean set an absolute directory?

**Reply by gajus** [reply]
```
- name: check conventions
  run: rp ...
  shell: 'script --return --quiet --command "bash {0}"'
```

so this works... though seems an overkill.

**Reply by BurntSushi** [reply]
> What do you mean by:
> 
> > So give the directory to your second command.
> 
> As far as I can tell, all commands specify the directory (the dot at the end). Do you mean set an absolute directory?

No. Your second command doesn't specify a directory to search. Look again. It it searching for the pattern `.` where as your first command searches for `test`.

**Reply by gajus** [reply]
I am confused. I only want to search by file name though. Does this mean I need to provide a dummy search phrase? e.g. `rg -g 'foo' '' .`?

**Reply by BurntSushi** [reply]
ripgrep is a tool that searches through the contents of files. It has numerous options for filtering by file path, but there is no "search only the file name." I'm confused as to why you thought there was. Which documentation materials told you otherwise?

Probably a pattern that always matches (the empty string will do, as you point out) is a work-around for your problem.

**Reply by gajus** [reply]
That's how I always used ag, and just migrated to rg. For what it is worth, it does the job. Thank you for the input!
